libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/RustTimeline.kt

```diff
@@ -368,17 +365,9 @@ class RustTimeline(
     ): Result<Unit> = withContext(dispatcher) {
         runCatching {
             val msg = messageEventContentFromParts(body, htmlBody).withMentions(mentions.map())
-            if (fromNotification) {
+            inner.sendReply(msg, eventId.value)
+            if (!fromNotification) {
```

Comment:
I think `specialModeEventTimelineItem` can be removed and the code can be cleaned up.

Reply:
https://github.com/element-hq/element-x-android/pull/3068/commits/55cd587fdf108db73c367aa8a0f98d9d232c7669